crates/nu-cli/src/commands/headers.rs

```diff
@@ -0,0 +1,76 @@
+use crate::commands::WholeStreamCommand;
+use crate::context::CommandRegistry;
+use crate::prelude::*;
+use futures::stream::StreamExt;
+use indexmap::IndexMap;
+use nu_errors::ShellError;
+use nu_protocol::Dictionary;
+use nu_protocol::{ReturnSuccess, Signature, UntaggedValue, Value};
+
+pub struct Headers;
+#[derive(Deserialize)]
+pub struct HeadersArgs {}
+
+impl WholeStreamCommand for Headers {
+    fn name(&self) -> &str {
+        "headers"
+    }
+    fn signature(&self) -> Signature {
+        Signature::build("headers")
+    }
+    fn usage(&self) -> &str {
+        "Use the first row of the table as headers"
+    }
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        args.process(registry, headers)?.run()
+    }
+}
+
+pub fn headers(
+    HeadersArgs {}: HeadersArgs,
+    RunnableContext { input, .. }: RunnableContext,
+) -> Result<OutputStream, ShellError> {
+    let stream = async_stream! {
+        let rows: Vec<Value> = input.values.collect().await;
+
+        //the headers are the first row in the table
+        let headers: Vec<String> = match &rows[0].value {
```

Comment:
This line might be causing the crash if it's given an empty stream.  You might want to do:

```
if let Some(headers) = rows.nth(0) {
  // ...
} else {
  return Err(ShellError(...));
}
```

Reply:
That was it!
I tried to go the `if let` route but I couldn't get it working as I wanted to, so I just added a short check above all the logic in order to keep it simple. Hope that's ok.

Thank you for all your comments and help, I really appreciate it!